src/Database/Database.php

```diff
@@ -605,13 +607,16 @@ public function p($sql)
 
 				$param_types = '';
 				$values      = [];
-				foreach ($args as $param => $value) {
+				foreach (array_keys($args) as $param) {
 					if (is_int($args[$param])) {
 						$param_types .= 'i';
 					} elseif (is_float($args[$param])) {
 						$param_types .= 'd';
 					} elseif (is_string($args[$param])) {
 						$param_types .= 's';
+					} elseif (is_object($args[$param]) && method_exists($args[$param], '__toString')) {
```

Comment:
Is this conversion needed when there is the casting from line 557?

Reply:
Line 557 is the PDO case, this is the mysqli case.

Reply:
Ah okay.